### PR TITLE
fix: empty document array not none

### PR DESCRIPTION
**Pull request by florian-hoenicke**

### CODE REVIEW

File: jina/peapods/runtimes/zmq/zed.py [hunk 1]
```diff
@@ -389,10 +389,12 @@ def _get_docs(self, field: str) -> 'DocumentArray':
...
             result = getattr(self.request, field)
 
-        # to unify all length=0 DocumentArray (or any other results) will simply considered as None
-        # otherwise the executor has to handle DocArray(0)
-        if len(result):
-            return result
+        # to unify all length=0 DocumentArray (or any other results) will simply considered as []
+        # otherwise the executor has to handle None values
+
+        if result is None:
```
Comment (JoanFM) [inline jina/peapods/runtimes/zmq/zed.py:395]: I would do it on client side. it is good to have the option to have docs as None (as Dump) or whatever, and it is what someone would expect if None is in inpus

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/peapods/runtimes/zmq/zed.py (hunk 1 above)
Comment (JoanFM) [reply jina/peapods/runtimes/zmq/zed.py:395]: Otherwise the condition of empty batch vs on purpose None inputs cannot be separated

Comment (florian-hoenicke) [reply jina/peapods/runtimes/zmq/zed.py:395]: I what file would you put it?

Comment (JoanFM) [reply jina/peapods/runtimes/zmq/zed.py:395]: Look for this function:

```python
U need to fix most likely this function:
 def _add_docs_groundtruths(req, batch, data_type, _kwargs):
     for content in batch:
         if isinstance(content, tuple) and len(content) == 2:
             # content comes in pair,  will take the first as the input and the second as the ground truth
 
             # note how data_type is cached
             d, data_type = _new_doc_from_data(content[0], data_type, **_kwargs)
             gt, _ = _new_doc_from_data(content[1], data_type, **_kwargs)
             req.docs.append(d)
             req.groundtruths.append(gt)
         else:
             d, data_type = _new_doc_from_data(content, data_type, **_kwargs)
             req.docs.append(d)
```

Comment (JoanFM) [reply jina/peapods/runtimes/zmq/zed.py:395]: I think u need to consider when batch is empty

Comment (JoanFM) [reply jina/peapods/runtimes/zmq/zed.py:395]: not 100% sure

Comment (florian-hoenicke) [reply jina/peapods/runtimes/zmq/zed.py:395]: I think the logic of adding the groundtruth should be separated from `None` or [] decisions

**Review by Gamol123 — APPROVED** [review]

File: jina/peapods/runtimes/zmq/zed.py (hunk 1 above)
Comment (JoanFM) [reply jina/peapods/runtimes/zmq/zed.py:395]: Also it poses the problem that is cheating the user. This new DocArray p

Comment (Kelton8Z) [reply jina/peapods/runtimes/zmq/zed.py:395]: maybe instruct users to do `bool(docArray)` to check if the document array is empty ?

**Comment by nan-wang** [comment]
https://jinaai.slack.com/archives/C018F60RBL5/p1626335308267700

**Pull request closed without merging**